Improve speed linting

simon-debruijn opened the pull request:
Context:

My vscode environment has become really slow at times. It seems to be the eslint that runs on save.
I started this PR to improve the speed of our linting without losing out on features.

These are my findings using the `TIMING=1 yarn lint` command

**Before change:**

```
Rule                                      | Time (ms) | Relative
:-----------------------------------------|----------:|--------:
@typescript-eslint/no-misused-promises    |  3368.998 |    34.5%
@typescript-eslint/promise-function-async |  3061.135 |    31.3%
@typescript-eslint/no-implied-eval        |  1388.643 |    14.2%
import/no-duplicates                      |   527.297 |     5.4%
react/display-name                        |   106.398 |     1.1%
simple-import-sort/imports                |    91.702 |     0.9%
react/no-deprecated                       |    86.960 |     0.9%
@typescript-eslint/no-base-to-string      |    84.452 |     0.9%
react/no-direct-mutation-state            |    70.299 |     0.7%
no-unused-vars                            |    61.109 |     0.6%
✨  Done in 19.98s.
```

**After change:**

```
@typescript-eslint/no-unused-vars |   296.503 |    24.9%
react/display-name                |   121.027 |    10.1%
react/no-deprecated               |   117.768 |     9.9%
simple-import-sort/imports        |   111.195 |     9.3%
react/no-direct-mutation-state    |    92.569 |     7.8%
react/no-string-refs              |    57.236 |     4.8%
react/require-render-return       |    50.311 |     4.2%
react/no-unknown-property         |    30.897 |     2.6%
no-redeclare                      |    29.949 |     2.5%
react-hooks/exhaustive-deps       |    20.351 |     1.7%
✨  Done in 7.45s.
```

brampauwelyn commented on package.json:
```diff
@@ -54,12 +52,10 @@
     "react-router": "^5.2.0",
     "react-table": "^7.6.3",
     "sass": "^1.27.0",
-    "sass-loader": "^10.0.4",
```
Is the sass-loader not necessary anymore?

simon-debruijn replied:
No since we changed the import for bootstrap to be via a cdn

brampauwelyn replied:
But we still need the `sass` pacakge?

brampauwelyn reviewed (approved):
Looks good to me. But @geoffreydhuyvetters  is probably in a better position to judge this.

geoffreydhuyvetters commented:
Could you explain in the description what the exact strategy is here? Or where this is based upon?

simon-debruijn commented:
> Could you explain in the description what the exact strategy is here? Or where this is based upon?

I removed the eslint config we had, started from scratch with the `yarn eslint --init` command and started adding extends and plugins along the way.

The way typescript is checked, I think, has the biggest influence on the performance of our linting.
 
I could re-add the `standard` rules we had, from their website it seemed like they really want you to use standard instead of eslint so that's why I left them out for now.

I also did a `depcheck` and removed some unused dependencies.
